contracts/helpers/MultiBlockRebalancer.sol

```diff
@@ -0,0 +1,313 @@
+     * @param maxExposureYieldAsset maximum exposure within the Transmuter to the asset
+     * @param overrideExposures whether limit exposures should be overriden or read onchain through the Transmuter
+     * This value should be 1 to override exposures or 2 if these shouldn't be overriden
+     */
+    function setCollateralData(
+        address collateral,
+        uint64 targetExposure,
+        uint64 minExposureYieldAsset,
+        uint64 maxExposureYieldAsset,
+        uint64 overrideExposures
+    ) external onlyGuardian {
+        CollatParams storage collatInfo = collateralData[collateral];
+        if (targetExposure >= 1e9) revert InvalidParam();
+        collatInfo.targetExposure = targetExposure;
+        collatInfo.overrideExposures = overrideExposures;
+        if (overrideExposures == 1) {
+            if (maxExposureYieldAsset >= 1e9 || minExposureYieldAsset >= maxExposureYieldAsset) revert InvalidParam();
+            collatInfo.maxExposureYieldAsset = maxExposureYieldAsset;
+            collatInfo.minExposureYieldAsset = minExposureYieldAsset;
+        } else {
+            collatInfo.overrideExposures = 2;
+            _updateLimitExposuresYieldAsset(collateral, collatInfo);
+        }
+    }
+
+    /**
+     * @notice Set the max allowed slippage
+     * @param newMaxSlippage new max allowed slippage
+     */
+    function setMaxSlippage(uint96 newMaxSlippage) external onlyGuardian {
+        _setMaxSlippage(newMaxSlippage);
+    }
+
+    /**
+     * @notice Set the deposit address for a collateral
+     * @param collateral address of the collateral
+     * @param newDepositAddress address to deposit to receive collateral
+     */
+    function setCollateralToDepositAddress(address collateral, address newDepositAddress) external onlyGuardian {
+        collateralToDepositAddress[collateral] = newDepositAddress;
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+                                                        TRUSTED FUNCTIONS
+    //////////////////////////////////////////////////////////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Initiate a rebalance
+     * @param scale scale to apply to the rebalance amount
+     * @param collateral address of the collateral
+     */
+    function initiateRebalance(uint256 scale, address collateral) external onlyTrusted {
+        if (scale > 1e9) revert InvalidParam();
+        (uint8 increase, uint256 amount) = _computeRebalanceAmount(collateral);
+        amount = (amount * scale) / 1e9;
+
+        try transmuter.updateOracle(collateral) {} catch {}
+        _rebalance(increase, collateral, amount);
+    }
+
+    /**
+     * @notice Finalize a rebalance
+     * @param collateral address of the collateral
+     */
+    function finalizeRebalance(address collateral) external onlyTrusted {
```

Comment:
actually maybe we can just specify the amount in the parameters, it will be capped by the balance amount anyway

Reply:
What would be the upsides of this approach ? as I don't really see why we would wants to only convert a specific amount ?